test/highlight/scala3.scala

```diff
@@ -23,6 +23,14 @@ object O1:
   //^keyword.function
   //  ^method
 
+object O2:
+  type Elem[A] = A match
+  //^keyword
+  //   ^type.definition
+    case String   => Char
```

Comment:
Do the actual types (both on left and right of =>) have highlighting on them? They should, so may be it's worth adding them to the test explicitly.